### ISSUE TITLE
webui: searching for "match" doesn't provide any results

### ISSUE THREAD

**Issue by mr-tz** [mandiant/capa#2349]
![image](https://github.com/user-attachments/assets/b712a5e0-cd01-4f5b-80f7-ebf08ba5108c)

I'd expect this to show this rule for example

![image](https://github.com/user-attachments/assets/e33ab5ac-c769-41a7-8799-a76fcaa1ebbf)

Or is this by design?

**Comment by s-ff** [comment]
>Or is this by design?

@mr-tz, yes this works as designed.

For example, a simple rule match looks like this:
```JSON
[
  {
    "key": "53-0",
    "data": {
      "type": "match location",
      "name": "function @ 0x1400198F7"
    },
    "children": [
      {
        "key": "53-0",
        "data": {
          "type": "statement",
          "typeValue": "or",
          "name": "or:",
        },
        "children": [
          {
            "data": {
              "type": "feature",
              "typeValue": "api",
              "name": "GlobalMemoryStatusEx",   <---
              "address": "0x1400344AE"
            },
            "children": []
          }
        ]
      }
    ]
  }
]
```

The search only operates on the `name` field and using templating, we add in `- ` and the `typeValue` (e.g. `api`, `count`, .. etc) which results in the table cell showing `- api: GlobalMemoryStatusEx` but actually it's just the `name` in the backend.

If we want to include the `typeValue` (e.g. `api`, `match`, ... etc), then the constructed `name` should be `typeValue: name`, e.g.
```JSON
{
            "data": {
              "type": "feature",
              "typeValue": "api",
              "name": "api: GlobalMemoryStatusEx",   <---
              "address": "0x1400344AE"
            },
            "children": []
}
```

Similarly, if we want to include the `-` in the search, and not via templating, then:

```JSON
{
            "data": {
              "type": "feature",
              "typeValue": "api",
              "name": "- api: GlobalMemoryStatusEx",   <---
              "address": "0x1400344AE"
            },
            "children": []
}
```
In conclusion, the search works on the JSON, and the templating is used to add in colors, text, styling, ... etc.

I hope this clears out things for you. Let me know if it is preferred to include these value and I can send a fix PR :)

**Comment by mr-tz** [comment]
It feels a little inconsistent. I can search for `optional` or `or` but not for `match` or `regex`.
These are different items though and your explanation makes sense.
